blocks/logic.js

```diff
@@ -274,8 +274,7 @@ Blockly.defineBlocksWithJsonArray([ // Mutator blocks. Do not extract.
  * Tooltip text, keyed by block OP value. Used by logic_compare and
  * logic_operation blocks.
  *
- * Messages are not dereferenced here in order to capture possible language
- * changes.
+ * Messages references are replaced just before displayed.
```

Comment:
Message references (`%{BKY...}`) are replaced just before showing the tooltip, thus the lookup into the string table occurs much later. This allows the message strings and this tooltip table to load in either order. It also allows the message table to change (change language) in a way that will be picked up by the dynamic tooltip.

My attention to this probably has a lot to do with unravelling the above issues in earlier PRs.